packages/core-events/director.ts

```diff
@@ -0,0 +1,41 @@
+import EventEmitter from 'events';
+
+export abstract class EventAdapter {
+  public abstract publish(eventName: string, payload: any): void;
+
+  public abstract subscribe(eventName: string, callBack: () => void): void;
+}
+
+class EventDirector {
+  private static adapter: EventAdapter;
+
+  static setEventAdapter(adapter: EventAdapter): void {
+    EventDirector.adapter = adapter;
+  }
+
+  static emit(eventName: string, payload: any): void {
+    EventDirector.adapter.publish(eventName, payload);
+  }
+
+  static subscribe(eventName: string, callBack: () => void): void {
+    EventDirector.adapter.subscribe(eventName, callBack);
+  }
+}
+
+class EventPlugins extends EventAdapter {
```

Comment:
So this is the actual adapter using Node's EventEmitter?

Reply:
yes, default support